src/V3Control.cpp

```diff
@@ -259,6 +259,17 @@ class V3ControlModule final {
         }
     }
 
+    void applyBlock(AstGenBlock* nodep) {
+        const VPragmaType pragma = VPragmaType::COVERAGE_BLOCK_OFF;
+        if (!nodep->unnamed()) {
+            for (const string& i : m_coverageOffBlocks) {
+                if (VString::wildmatch(nodep->prettyOrigOrName(), i)) {
+                    nodep->addStmtsp(new AstPragma{nodep->fileline(), pragma});
```

Comment:
Now that it's separate code please check it and the code below has some .vlt test coverage for these, as it might not have been covered previously since overlapped normal non-gen begin coverage.